src/server/shared/Updater/DBUpdater.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2008-2015 TrinityCore <http://www.trinitycore.org/>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef DBUpdater_h__
+#define DBUpdater_h__
+
+#include <vector>
+
+#include "DatabaseWorkerPool.h"
+
+class DBUpdater
+{
+    DBUpdater() { }
+    ~DBUpdater() { }
+
+public:
+    static DBUpdater* instance()
+    {
+        static DBUpdater instance;
+        return &instance;
+    }
+
+    template<class T>
+    uint32 update(DatabaseWorkerPool<T> const& pool, std::string const& path);
+
+private:
+    DBUpdater(DBUpdater const&);
+    DBUpdater& operator=(DBUpdater const&);
+
+    std::vector<std::string> getFileList(boost::filesystem::path const& updatedir);
```

Comment:
Nah, we have always used UpperCamelCase (http://collab.kpsn.org/pages/viewpage.action?pageId=10682946#C++DevelopmentStandards-Namingofvariablesandfunctions)

Anyway, it might be worth if we spend more time in research/discussing the feature before jumping right into the code. Can you join us in our IRC channel? (#trinity @ irc.rizon.net)